### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by augis10**
Registration with authentication done.
To login use:
test2@test.lt
test

create account: 
Post - > http://localhost:8080/api/v1/registration
Payload -> 

{
    "firstName": "test",
    "lastName": "test",
    "email": "test5@test.lt",
    "password": "test"
}

email must be unused

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -76,6 +76,21 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>28.2-jre</version>
+        </dependency>
+
```
Comment (Bernardas-Labutis) [inline backend/pom.xml]: Nebuvo tarpu tarp ankstesniu dependency tag`u

File: backend/src/main/java/lt/vu/tube/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                .csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/api/v*/registration/**").permitAll()
+                .antMatchers("/", "index", "/css/*", "/js/*").permitAll()
+                .anyRequest()
+                .authenticated().and()
+                .formLogin().defaultSuccessUrl("/videos")
+                .and()
+                .logout()
+                    .logoutUrl("/logout")
```
Comment (Bernardas-Labutis) [inline backend/src/main/java/lt/vu/tube/config/WebSecurityConfig.java]: formatavimas

File: backend/src/main/java/lt/vu/tube/entity/AppUser.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+@Getter
+@Setter
+@EqualsAndHashCode
+@NoArgsConstructor
+@Entity
+public class AppUser implements UserDetails {
+
+
+    @SequenceGenerator(
+            name = "student_sequence",
```
Comment (Bernardas-Labutis) [inline backend/src/main/java/lt/vu/tube/entity/AppUser.java]: Ar mum reikia custominio sequenco? Gal galim naudot tiesiog hibernate`o standartini?

File: backend/pom.xml (hunk 1 above)
Comment (agosu) [inline backend/pom.xml]: random empty line

File: backend/src/main/java/lt/vu/tube/services/EmailValidator.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.springframework.stereotype.Service;
+
+import java.util.function.Predicate;
+
+@Service
+public class EmailValidator implements Predicate<String> {
+    @Override
+    public boolean test(String s) {
+//        TODO: Regex to validate email
```
Comment (Bernardas-Labutis) [inline backend/src/main/java/lt/vu/tube/services/EmailValidator.java]:
```
public static final Pattern VALID_EMAIL_ADDRESS_REGEX = 
    Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);

public static boolean validate(String emailStr) {
        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(emailStr);
        return matcher.find();
}
```

File: backend/src/main/java/lt/vu/tube/services/EmailValidator.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package lt.vu.tube.services;
+
+import org.springframework.stereotype.Service;
+
+import java.util.function.Predicate;
+
+@Service
+public class EmailValidator implements Predicate<String> {
+    @Override
+    public boolean test(String s) {
```
Comment (Bernardas-Labutis) [inline backend/src/main/java/lt/vu/tube/services/EmailValidator.java]: pervadink i 'isValid'

File: backend/src/main/java/lt/vu/tube/web/VideoController.java [hunk 6]
```diff
@@ -154,18 +157,24 @@ else if(neededParams.contains(item.getFieldName())) {
...
         return StreamSupport.stream(videoRepository.findAll().spliterator(), false).collect(Collectors.toList());
     }
 
     //Temporary delete later
     @RequestMapping("/videoLinks")
+    //@PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @PreAuthorize("hasAuthority('user:read')")
     public Map<UUID, String> getVideoLinks() throws IOException {
         return StreamSupport.stream(videoRepository.findAll().spliterator(), false)
             .collect(Collectors.toMap(Video::getId, v -> {
```
Comment (Bernardas-Labutis) [inline backend/src/main/java/lt/vu/tube/web/VideoController.java:180]: Negalėjau čia pakomentuoti, bet toks jausmas kad žemiau sucommitinti failai yra iš target`o, jų neturėtų čia būt

**Review by Bernardas-Labutis — CHANGES_REQUESTED** [review]

File: backend/src/main/java/lt/vu/tube/config/WebSecurityConfig.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.authenticationProvider(daoAuthenticationProvider());
+    }
+
+    @Bean
+    public DaoAuthenticationProvider daoAuthenticationProvider() {
+        DaoAuthenticationProvider provider =
+                new DaoAuthenticationProvider();
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/config/WebSecurityConfig.java]: why new line?

File: backend/src/main/java/lt/vu/tube/config/WebSecurityConfig.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+@AllArgsConstructor
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    private final AppUserService appUserService;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/config/WebSecurityConfig.java:24]: more usual http.csrf().disable()

File: backend/src/main/java/lt/vu/tube/entity/AppUser.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.Collection;
+import java.util.Collections;
+
+@Getter
+@Setter
+@EqualsAndHashCode
+@NoArgsConstructor
+@Entity
+public class AppUser implements UserDetails {
+
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/entity/AppUser.java]: empty line

File: backend/src/main/java/lt/vu/tube/entity/AppUser.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+            generator = "student_sequence"
+    )
+    private Long id;
+    private String firstName;
+    private String lastName;
+    private String email;
+    private String password;
+    @Enumerated(EnumType.STRING)
+    private AppUserRole appUserRole;
+    private Boolean locked = false;
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/entity/AppUser.java]: why use capital Boolean? boolean saves a lot of memory. Unless we really need Boolean object methods

File: backend/src/main/java/lt/vu/tube/enums/AppUserRole.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public Set<SimpleGrantedAuthority> getGrantedAuthorities() {
+        Set<SimpleGrantedAuthority> permissions = getPermissions().stream()
+                .map(permission -> new SimpleGrantedAuthority(permission.getPermission()))
+                .collect(Collectors.toSet());
+        permissions.add(new SimpleGrantedAuthority("ROLE_" + this.name()));
+        return permissions;
+    }
+}
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/enums/AppUserRole.java]: git complains about no new line

File: backend/src/main/java/lt/vu/tube/enums/AppUserRole.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import static lt.vu.tube.enums.ApplicationUserPermission.*;
+
+public enum AppUserRole {
+    USER(Sets.newHashSet(USER_READ, USER_WRITE)),
+    ADMIN(Sets.newHashSet(USER_READ, USER_WRITE));
+
+    private final Set<ApplicationUserPermission> permisions;
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/enums/AppUserRole.java]: typo here and in 17, 18, 22.

File: backend/src/main/java/lt/vu/tube/repository/AppUserRepository.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Repository
+@Transactional(readOnly = true)
+public interface AppUserRepository
+        extends JpaRepository<AppUser, Long> {
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/repository/AppUserRepository.java]: 80 symbols is very far away in IntelliJ, why split into two lines?

File: backend/src/main/java/lt/vu/tube/requests/RegistrationRequest.java [hunk 14]
```diff
@@ -0,0 +1,17 @@
+package lt.vu.tube.requests;
+
+import lombok.*;
+
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/requests/RegistrationRequest.java:4]: empty line

File: backend/src/main/java/lt/vu/tube/requests/RegistrationRequest.java [hunk 15]
```diff
@@ -0,0 +1,17 @@
+package lt.vu.tube.requests;
+
+import lombok.*;
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/requests/RegistrationRequest.java:3]: We decided to not use lombok?

File: backend/src/main/java/lt/vu/tube/services/AuthenticatedUser.java [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+
+import lt.vu.tube.entity.AppUser;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticatedUser {
+    public static AppUser getAuthenticatedUser(){
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if(principal instanceof AppUser) {
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/services/AuthenticatedUser.java]: good practice to leave space after if and (, same with else

File: backend/src/main/java/lt/vu/tube/services/RegistrationService.java [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+    public String register(RegistrationRequest request) {
+        boolean isValidEmail = emailValidator.
+                test(request.getEmail());
+
+        if (!isValidEmail) {
+            throw new IllegalStateException("email not valid");
+        }
+
+        String response = appUserService.signUpUser(
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/services/RegistrationService.java]: IntelliJ says local variable is redundant, you can directly return it

File: backend/src/main/java/lt/vu/tube/web/HelloController.java [hunk 18]
```diff
@@ -19,7 +21,12 @@
 
     @RequestMapping("/")
     public String index() {
-        return "Greetings from Spring Boot!";
+        AppUser user = AuthenticatedUser.getAuthenticatedUser();
+        String username = "Anonymous Person";
+        if(user != null){
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/web/HelloController.java]: if (

**Review by agosu — DISMISSED** [review]
We don't need to commit jar files. Maybe gitignore should be updated

**Review by Bernardas-Labutis — APPROVED** [review]

**Pull request merged**